### PR TITLE
PIM-3746: Avoid hard coded classes in form type inject them in services

### PR DESCRIPTION
**Pull request by solivier**
| Q                    | A
| -------------------- | ---
| Bug fix?             | No
| New feature?         | No
| BC breaks?           | Yes
| CI currently passes? | Yes
| Tests pass?          | Yes
| Scenarios pass?      | Yes
| Checkstyle issues?*  | No
| PMD issues?**        | No
| Changelog updated?   | yes
| Fixed tickets        | PIM-3746
| DB schema updated?   | No
| Migration script?    | No
| Doc PR               | -

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/config/form_types.yml [hunk 1]
```diff
@@ -130,12 +140,15 @@ services:
             - { name: form.type, alias: pim_enrich_family }
         arguments:
             - '@pim_enrich.form.subscriber.attribute_requirements'
-            - %pim_catalog.entity.attribute.class%
             - '@pim_enrich.form.subscriber.family_disable_fields_subscriber'
             - '@pim_enrich.form.subscriber.family_attribute_label_subscriber'
+            - %pim_catalog.entity.attribute.class%
+            - '%pim_catalog.entity.family.class%'
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Resources/config/form_types.yml]: no '%

File: src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/EditCommonAttributesType.php [hunk 2]
```diff
@@ -19,34 +19,40 @@
...
      * @param string                   $attributeClass
+     * @param string                   $className
+     * @param string                   $dataClass
      */
     public function __construct(
         ProductFormViewInterface $productFormView,
         LocaleHelper $localeHelper,
-        $attributeClass
+        $attributeClass,
+        $className,
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/EditCommonAttributesType.php]: not really explicit between classname and dataclass

File: src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/ClassifyType.php [hunk 3]
```diff
@@ -18,12 +18,17 @@ class ClassifyType extends AbstractType
...
 
+    /** @var string */
+    protected $dataClass;
+
     /**
      * @param string $categoryClass
+     * @param string $dataClass
      */
-    public function __construct($categoryClass)
+    public function __construct($categoryClass, $dataClass)
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/ClassifyType.php]: not explicit

Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/ClassifyType.php]: actionClass ?

File: src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/AddToVariantGroupType.php [hunk 4]
```diff
@@ -15,6 +15,22 @@
...
+    protected $dataClass;
+
+    /** @var string */
+    protected $className;
+
+    /**
+     * @param string $className
+     * @param string $dataClass
+     */
+    public function __construct($className, $dataClass)
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/AddToVariantGroupType.php]: not explicit

File: src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/AddToGroupsType.php [hunk 5]
```diff
@@ -15,6 +15,22 @@
...
+    protected $dataClass;
+
+    /** @var string */
+    protected $className;
+
+    /**
+     * @param string $className
+     * @param string $dataClass
+     */
+    public function __construct($className, $dataClass)
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Type/MassEditAction/AddToGroupsType.php]: not explicit

File: src/Pim/Bundle/EnrichBundle/Form/Type/GroupType.php [hunk 6]
```diff
@@ -30,16 +30,21 @@ class GroupType extends AbstractType
...
+
     /**
      * Constructor
      *
      * @param ProductRepositoryInterface $productRepository
      * @param string                     $attributeClass
+     * @param string                     $dataClass
      */
-    public function __construct(ProductRepositoryInterface $productRepository, $attributeClass)
+    public function __construct(ProductRepositoryInterface $productRepository, $attributeClass, $dataClass)
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Type/GroupType.php]: not explicit

File: spec/Pim/Bundle/EnrichBundle/Form/Type/AssociationTypeSpec.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    {
+        $this->shouldBeAnInstanceOf('Symfony\Component\Form\AbstractType');
+    }
+
+    function it_has_a_name()
+    {
+        $this->getName()->shouldReturn('pim_enrich_association');
+    }
+
+    function it_build_form(
```
Comment (sumbobyboys) [inline spec/Pim/Bundle/EnrichBundle/Form/Type/AssociationTypeSpec.php]: builds

**Pull request merged**